webapp/src/ts/services/enketo.service.ts

```diff
@@ -309,6 +290,10 @@ export class EnketoService {
           return Promise.reject(new Error(JSON.stringify(loadErrors)));
         }
       })
+      .then(() => this.languageService.get())
+      // TODO remove this when our enketo-core dependency is updated as the latest
+      //      version uses the language passed to the constructor
+      .then((language) => this.currentForm.langs.$formLanguages.val(language).trigger('change'))
```

Comment:
Unfortunately, this doesn't work. 
This function calls `$formLanguages.setAll` with the passed language, which is what we used to do ourselves as well, here: https://github.com/medic/cht-core/blame/a2d6696f9f73b2d3006021a500df5ff782b00ee9/webapp/src/js/services/enketo.js#L259

It was not working for cascading repeats:
https://github.com/medic/cht-core/issues/6288
https://github.com/medic/cht-core/issues/6279

If you use the sample form that Derick provided in those issues and try them on your branch, you'll see the regression. 

your branch: 
![image](https://user-images.githubusercontent.com/35681649/127447963-3a016352-2038-414d-9d65-86b46b6fc15b.png)

master: 
![image](https://user-images.githubusercontent.com/35681649/127448139-02e9da50-720e-4653-8dd7-7f7fc34a3892.png)

Reply:
Nice catch! Enketo keeps the repeat templates as is, with fields in English by default.  
You're right, my solution doesn't cover this case and we must know when something like this breaks.

I'm still figuring out why Enketo doesn't pick up the current language and unless I can make Enketo work with it, I will put back the `setAll` function body in our code.

Reply:
When Enketo initializes a form, it will save the repeat templates in memory and remove them from the DOM. This happens before we tell Enketo the user's language so the templates are saved in the default language which is English.  
When we set the language,the `setAll` method is called but the templates have already been removed from the DOM and saved in memory in English. Which is why the repeat fields are currently shown in English.

This is definitely something that should be fixed by upgrading Enketo. In the latest version, the form's language is initialized in `EnketoForm` constructor through a dedicated parameter.

Reply:
An alternative approach I have found to work is to re-set Enketo's language to the user's language when adding a repeat element like this:
```typescript
const setLanguage = () => this.currentForm.langs.$formLanguages.val(language).trigger('change');
this.currentForm.view.$.on('click', 'button.add-repeat-btn:enabled', setLanguage);
```

This solution works for this specific use case raised by Derick. Is there another use case that would warrant bringing back the `setAll` method body in our code?

I'm not attached to either solution. Objectively, bringing back the `setAll` method body in our code seems like the best one for now.  
What do you think @dianabarsan ?

Reply:
If that works, then it's great. 
Can we maybe shorten it and use: 
```
this.currentForm.langs.setAll(language);
```
instead of 
```
this.currentForm.langs.$formLanguages.val(language).trigger('change');
```
? 

Does it make a different to set Enketo's language?

Reply:
We have to do it at least once because it sets the form's current language (https://github.com/enketo/enketo-core/blob/4.41.6/src/js/language.js#L39)  
And this is what fixes the initial #6964 issue